velox/functions/prestosql/ArithmeticImpl.h

```diff
@@ -161,4 +161,46 @@ FOLLY_ALWAYS_INLINE double truncate(
   }
 }
 
+// helper function for calculating upper and lower limit of wilson interval
+FOLLY_ALWAYS_INLINE void wilsonInterval(
```

Comment:
I glanced over the implementation and it differs to the Java implementation.
The comment indicates it is done handle issues with INFINITY.

At the same time, both, the upper and lower bound are always calculated, when only one of these is actually neded. Wondering if we compute more than we actually need to and if this can be changed a bit.

Reply:
In addition, the Java tests has tests for these values
```
assertFunction("wilson_interval_lower(1250, 1310, 1.96e0)", DOUBLE, 0.9414883725395894);
assertFunction("wilson_interval_upper(1250, 1310, 1.96e0)", DOUBLE, 0.9642524717143908);
```

perhaps these could be added as well.
This PR has more tests than the Java code which is nice.

Reply:
I've updated the code so that both upper and lower bound are no longer calculated during the same call. I've done this using a boolean template argument - the `if (upper)` branches can be resolved at compile time and should be optimised out. Similarly for the calculation of `a` and `c`.

I've also added the two extra Java test cases.